### PR TITLE
Basics of forwarding a POST to /api/v0/events to kafka

### PR DESCRIPTION
**Pull request by steveburkett**
Created a basic controller action (POST to /api/v0/events).  There is NO data/schema validation, this is just v0 of the API

https://app.zenhub.com/workspaces/quasar-5ef270981e560b0019803111/issues/openstax/quasar/9

```
{
	"event": {
		"data": "brocolli5",
		"topic": "rails-test"
	}
}
```

image

image

### CODE REVIEW

File: app/controllers/api/v0/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Api::V0::EventsController < ApplicationController
+  def create
+    KafkaClient.produce(data: params[:event][:data], topic: params[:event][:topic])
+    render json: { message: 'created' }, status: :ok
```
Comment (TomWoodward) [inline app/controllers/api/v0/events_controller.rb]: you could probably just return an empty 201

File: app/lib/kafka_client.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                    delivery_interval: Rails.application.secrets.kafka[:delivery_interval]
+                  )
+      end
+  end
+
+  def self.produce(data:, topic:)
+    kafka_producer.produce(data, topic: topic)
+
+    # kafka_producer.deliver_messages  (this forces delivery to happen)
+    # when is shutdown needed?
```
Comment (TomWoodward) [inline app/lib/kafka_client.rb]: what is this about and should it be addressed in this pr?

File: config/secrets.yml.example [hunk 3]
```diff
@@ -0,0 +1,21 @@
+development:
```
Comment (TomWoodward) [inline config/secrets.yml.example:1]: please add this line from highlights api to the event api dockerfile https://github.com/openstax/highlights-api/blob/master/Dockerfile#L11 to create the secrets.yml

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: app/lib/kafka_client.rb (hunk 2 above)
Comment (steveburkett) [reply app/lib/kafka_client.rb]: @TomWoodward i left this there as a reminder to find a place to do shutdown properly.  found one. 

https://apidock.com/ruby/Kernel/at_exit

File: app/lib/kafka_client.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class KafkaClient
+  def self.produce(data:, topic:)
+    $kafka_producer.produce(data, topic: topic)
```
Comment (TomWoodward) [inline app/lib/kafka_client.rb]: 😱

File: config/initializers/kafka.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+require 'kafka'
+
+if Rails.env.development?
+  ActiveSupport::Notifications.subscribe(/.*\.kafka$/) do |*args|
+    event = ActiveSupport::Notifications::Event.new(*args)
+    puts "Received Kafka notification `#{event.name}` with payload: #{event.payload.inspect}"
+  end
+end
+
+$kafka = Kafka.new([Rails.application.secrets.kafka[:broker_host]],
```
Comment (TomWoodward) [inline config/initializers/kafka.rb]: does this have to be global?

**Review by TomWoodward — APPROVED** [review]

File: app/controllers/api/v0/events_controller.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Api::V0::EventsController < ApplicationController
+  def create
+    KafkaClient.produce(data: params[:event][:data], topic: params[:event][:topic])
```
Comment (TomWoodward) [inline app/controllers/api/v0/events_controller.rb:5]: do you have a plan in your head for something other than this `event` wrapper at the top level? we might want to just get rid of that

Comment (steveburkett) [reply app/controllers/api/v0/events_controller.rb:5]: @TomWoodward for sure i did...i figured that the KafkaClient will encapsulate a lot of stuff down the road, like schema validations, logging, exception handling, and prob. more.

**Pull request merged**